data_analysis/plots.py

```diff
@@ -1213,3 +1218,185 @@ def make_descriptive_plots(
         f"{start_date}_to_{ridership_end_date}"
         f"_all_day_types",
     )
+
+
+def calculate_ratio_per_ward(
+    ward_df: pd.DataFrame, summary_df: pd.DataFrame
+) -> pd.DataFrame:
+    """
+    Args:
+        ward_df (pd.DataFrame): DataFrame containing all routes
+            passing through a ward
+        summary_df (pd.DataFrame): DataFrame containing trip ratio per route
+
+    Returns:
+        pd.DataFrame: DataFrame containing median and weighted median
+            trip ratio of all routes per ward
+    """
+    # Deal with list values in 'routes' column
+    df_explode = ward_df.explode("routes").reset_index(drop=True)
+    df_explode = df_explode.drop(columns=["trip_count_rt", "trip_count_sched", "ratio"])
+    merged_df = summary_df.merge(df_explode, left_on="route_id", right_on="routes")
+    freq_df = (
+        merged_df["route_id"]
+        .value_counts()
+        .reset_index()
+        .rename(columns={"index": "route_id", "route_id": "count"})
+    )
+    merged_df = merged_df.merge(freq_df, on="route_id")
+    merged_df["weights"] = 1 / merged_df["count"]
```

Comment:
Just curious if we can comment this to explain a bit more, I am not sure what "count" means

Reply:
Count is the number of routes passing through a particular ward
 
```python
           ward  count
     1890    37     32
     2562    27     64
     1068    14     36
     1732    37     32
     2550    36     64
```